xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -329,29 +334,37 @@ void CGUIDialogAddonInfo::OnInstall()
   if (!g_passwordManager.CheckMenuLock(WINDOW_ADDON_BROWSER))
     return;
 
-  if (m_localAddon || !m_item->HasAddonInfo())
+  if (!m_item->HasAddonInfo())
     return;
 
-  std::string addonId = m_item->GetAddonInfo()->ID();
-  std::vector<std::pair<AddonVersion, std::string>> versions;
-
-  CAddonDatabase database;
-  if (!database.Open() || !database.GetAvailableVersions(addonId, versions) || versions.empty())
+  if (m_localAddon)
   {
-    CLog::Log(LOGERROR, "ADDON: no available versions of %s", addonId.c_str());
-    return;
-  }
+    const std::string& header = g_localizeStrings.Get(19098); // Warning!
+    const std::string& text =
+        StringUtils::Format(g_localizeStrings.Get(39028), m_localAddon->ID(),
```

Comment:
`Stringutils::Format`does not return a reference, but an object. This can crash! Does it even compile?

Reply:
this builds and shows the dialog correctly with header 'n' text.. change it anyway? if yes, to what?

Reply:
This works by luck only! Please change to `const std::string text =`